enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,508 @@
+metrics that expose whether a policy is compliant or noncompliant but those must be scraped by Prometheus and changes
+between scrapes will be lost, therefore, the data cannot be relied upon.
+
+When a user wants to pass an audit or determine when an incident occurred, it is required to have accurrate, historical,
+and queryable compliance data.
+
+### Goals
+
+1. Provide an accurate compliance history on the hub or other centralized location.
+1. The compliance history is stored in a queryable format.
+1. The complaince history can contain data for at least 2 years for a hub that manages 3000 managed clusters with 30
+   policies that mostly stay compliant.
+1. The compliance history retains data of deleted managed clusters and deleted policies.
+1. The outcome is that policy compliance events can be queried by a date range and at least the cluster name, parent
+   policy name, policy name, compliance state, and compliance message can be shown. The solution should allow extending
+   with more data.
+1. This feature should be opt-in.
+
+### Non-Goals
+
+1. Open Cluster Management will not manage the Postgresql database. The user must bring their own.
+
+## Proposal
+
+### Design
+
+#### Database Design
+
+The compliance history needs to be stored efficiently in a queryable format. Since the data format is consistent, using
+a relational database is a good fit. A proposed database schema is shown below. Note that this schema allows the same
+database to be used for multiple Open Cluster Management hubs, though if managed cluster names overlap, it'd be less
+confusing if a separate database on the same Postgresql database be used.
+
+##### clusters Table
+
+| Column     | Description                                                                                |
+| ---------- | ------------------------------------------------------------------------------------------ |
+| id         | an internal primary key                                                                    |
+| name       | a string column for the cluster name in the ManagedCluster object; this will have an index |
+| cluster_id | the ACM generated unique cluster ID; this will have a unique constraint                    |
+
+##### parent_policies Table
+
+| Column     | Description                                                         |
+| ---------- | ------------------------------------------------------------------- |
+| id         | an internal primary key                                             |
+| name       | a string column for the name of the parent replicated policy        |
+| categories | an array column of the sorted categories associated with the policy |
+| controls   | an array column of the sorted controls associated with the policy   |
+| standards  | an array column of the sorted standards associated with the policy  |
+
+There would be a combined unique constraint on the `name`, `categories`, `controls`, and `standards` columns. Note that
```

Comment:
What would happen if I update a policy, and add a new standard to it?

Reply:
There would be a new `parent_policies` row. That's what has been the tricky part in this database schema design is we need to store what things were in that point in time and not what they currently are, so it does lead to some data duplication.